share/spack/docker/ubuntu-2004.dockerfile

```diff
@@ -0,0 +1,76 @@
+FROM ubuntu:20.04
+MAINTAINER Spack Maintainers <maintainers@spack.io>
+
+ENV DOCKERFILE_BASE=ubuntu            \
+    DOCKERFILE_DISTRO=ubuntu          \
+    DOCKERFILE_DISTRO_VERSION=20.04   \
+    SPACK_ROOT=/opt/spack             \
+    DEBIAN_FRONTEND=noninteractive    \
+    CURRENTLY_BUILDING_DOCKER_IMAGE=1 \
+    container=docker
+
+RUN apt-get -yqq update \
+ && apt-get -yqq install --no-install-recommends \
+        build-essential \
+        ca-certificates \
+        curl \
+        file \
+        g++ \
+        gcc \
+        gfortran \
+        git \
+        gnupg2 \
+        iproute2 \
+        lmod \
+        locales \
+        lua-posix \
+        make \
+        python3 \
+        python3-pip \
+        python3-setuptools \
+        tcl \
+        unzip \
+ && locale-gen en_US.UTF-8 \
+ && pip3 install boto3 \
+ && rm -rf /var/lib/apt/lists/*
+
+COPY bin   $SPACK_ROOT/bin
+COPY etc   $SPACK_ROOT/etc
+COPY lib   $SPACK_ROOT/lib
+COPY share $SPACK_ROOT/share
+COPY var   $SPACK_ROOT/var
+RUN mkdir -p $SPACK_ROOT/opt/spack
+
+RUN ln -s $SPACK_ROOT/share/spack/docker/entrypoint.bash \
+          /usr/local/bin/docker-shell \
+ && ln -s $SPACK_ROOT/share/spack/docker/entrypoint.bash \
+          /usr/local/bin/interactive-shell \
+ && ln -s $SPACK_ROOT/share/spack/docker/entrypoint.bash \
+          /usr/local/bin/spack-env
+
+# Add LANG default to en_US.UTF-8
+ENV LANGUAGE en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LC_ALL en_US.UTF-8
+
+RUN mkdir -p /root/.spack \
+ && cp $SPACK_ROOT/share/spack/docker/modules.yaml \
+        /root/.spack/modules.yaml \
+ && rm -rf /root/*.* /run/nologin $SPACK_ROOT/.git
+
+RUN [ -f ~/.profile ]                                               \
+ && sed -i 's/mesg n/( tty -s \&\& mesg n || true )/g' ~/.profile \
+ || true
+
+# [WORKAROUND]
+# https://bugs.launchpad.net/ubuntu/+source/lua-posix/+bug/1752082
+RUN ln -s posix_c.so /usr/lib/x86_64-linux-gnu/lua/5.2/posix.so
```

Comment:
Are you able to build this image? For me it fails with:
```
Step 17/22 : RUN ln -s posix_c.so /usr/lib/x86_64-linux-gnu/lua/5.2/posix.so
 ---> Running in 1fe0af8be2ab
ln: failed to create symbolic link '/usr/lib/x86_64-linux-gnu/lua/5.2/posix.so': File exists
The command '/bin/sh -c ln -s posix_c.so /usr/lib/x86_64-linux-gnu/lua/5.2/posix.so' returned a non-zero code: 1
```
Maybe the bug was specific to Ubuntu 18.04.